src/main/java/com/assertthat/selenium_shutterbug/utils/web/Browser.java

```diff
@@ -134,12 +135,10 @@ public BufferedImage takeScreenshotEntirePage() {
     }
 
     public BufferedImage takeScreenshotEntirePageUsingChromeCommand() {
-        if(!(this.driver instanceof ChromeDriver))
-            throw new UnsupportedOperationException("You must use Chrome driver for this operation");
-
+        RemoteWebDriver remoteWebDriver = getDriverAfterValidation(this.driver);
         Browser driver;
```

Comment:
Lets not create new Browser instance inside another Browser instance.
Just reassign driver after validation:
```
this.driver = getDriverAfterValidation(this.driver);
```